src/content/migrate/5.md

```diff
@@ -0,0 +1,152 @@
+```
+
+W> webpack 5 removes all deprecated features. In order to proceed you must have no webpack deprecation warnings during the build.
+
+#### Make sure you are using entry point information from stats
+
+T> If you are using [HtmlWebpackPlugin](/plugins/html-webpack-plugin/) skip this step.
+
+When using static HTML or creating HTML in other way, make sure to use entry points from stats JSON to generate <script>, <style> and <link> tags.
+
+If this is not possible, avoid setting `splitChunks.chunks: 'all'` and `splitChunks.maxSize` later in this guide. Note that this is sub-optimal and a workaround.
+
+#### Make sure to use `mode`
+
+Set mode to either [`production`](/configuration/mode/#mode-production) or [`development`](/configuration/mode/#mode-development) to make sure that corresponding defaults set.
+
+#### Update outdated options
+
+Update the following options to their new version (if used):
+
+- `optimization.hashedModuleIds: true` => `optimization.moduleIds: 'hashed'`
+- `optimization.namedChunks: true` => `optimization.chunkIds: 'named'`
+- `optimization.namedModules: true` => `optimization.moduleIds: 'named'`
+- `NamedModulesPlugin` => `optimization.moduleIds: 'named'`
+- `NamedChunksPlugin` => `optimization.chunkIds: 'named'`
+- `HashedModulesPlugin` => `optimization.moduleIds: 'hashed'`
+- `optimization.occurrenceOrder: true` => `optimization: { chunkIds: 'total-size', moduleIds: 'size' }`
+
+#### Disable ES2015 syntax in runtime code, if necessary
+
+By default, webpack's runtime code uses ES2015 syntax to build smaller bundles. If your build target's environments that don't support this syntax (like IE11), you'll need to set `output.ecmaVersion: 5` to revert to ES5 syntax.
+
+#### Test webpack 5 compatibility
+
+Try to set the following options in your webpack 4 configuration and check if build still works correctly.
+
+```javascript
+module.exports = {
+  // ...
+  node: {
+    Buffer: false,
+    process: false
+  }
+};
+```
+
+T> webpack 5 removes these options from the configuration schema and will always use `false`. You have to remove these options again when upgrading your configuration for webpack 5.
+
+#### Upgrade webpack version
+
+NPM: `npm install webpack@next --dev`
+Yarn: `yarn add webpack@next -D`
+
+#### Clean up configuration
+
+- Consider removing `optimization.moduleIds` and `optimization.chunkIds` from your webpack configuration. The defaults could be better, because they support long term caching in [`production mode`](/configuration/mode/#mode-production) and debugging in [`development` mode](/configuration/mode/#mode-development)
+- When using `[hash]` placeholder in webpack configuration, consider changing it to `[contenthash]`. It is not the same, but proven to be more effective.
+- If you were using WASM and had `experiments.syncWebAssembly: true`, after migration to webpack 5, change it to `experiments: { asyncWebAssembly: true, importAsync: true }`.
```

Comment:
- If you were using WASM via `import` you should follow this two step process:
  - Enable the deprecated spec with `experiments.syncWebAssembly: true`, to get the same behavior as in webpack 4.
  - After successful migration to webpack 5, change it to `experiments: { asyncWebAssembly: true, importAsync: true }` to use the up-to-date spec for WASM integration.